src/chrome/update-shared-scope.ts

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+import { getSharedScope, initSharedScope } from '@scalprum/core';
+import { LinkProps, NavLinkProps, NavigateOptions, NavigateProps, To } from 'react-router-dom';
+
+const updateSharedScope = () => {
+  const calculateTo = (to: To) => {
+    if (window.location.pathname.includes('/hac')) {
+      // FIXME: Create a global dynamic plugin solution to scope plugin nested routes
+      if (typeof to === 'string' && !to.startsWith('/hac') && to.startsWith('/')) {
+        return `/hac${to}`;
+      } else if (typeof to !== 'string' && to.pathname && !to.pathname.startsWith('/hac')) {
+        return {
+          ...to,
+          pathname: `/hac${to.pathname}`,
+        };
+      }
+      return to;
+    }
+    return to;
+  };
+
+  const modules = {
+    'react-router-dom': async () => () => {
+      // We have to hack our way around react-router-dom
+      // Since we are no longer using basename we have to include `/hac` prefix
+      const reactRouter = require('react-router-dom');
+      return {
+        ...reactRouter,
```

Comment:
This is an interesting way to hack into / extend the existing shared modules 😃

Reply:
I am not a huge fan but for now it's the only way if we don't want to enforce code changes in existing hac plugins :upside_down_face: . We are hoping to come up with some generic plugin(extenstion) that would simulate the basename.

Reply:
I think this will replicate the same behavior/limitation we have today with HAC running on Chrome. We should have a larger discussion around our stance going forward on this. I don't think we're opposed to making code changes in HAC plugins - more. so do we want another explicit relationship between plugins and Chrome versus abstracting this functionality (whether that abstraction is an override, extension, utility, etc)